### PR TITLE
Major Re-Organizing

### PR DESCRIPTION
**Pull request by ellenkorbes**
Changes made:

The `fast/` directory contains the different stages presented in the "Quest For The Fastest Deployment Time" talk.

The `full/` directory contains the whole app, fully optimized, to be used on demos and such.

Notable functionality in the full version:

- All Go services re-compile locally and live reload the resulting binaries.
- Every time a change is made to a Go file in one of the services, that update gets benchmarked,
- All that's needed to do so is to add `Benchmark()` after `main() {`.
- Reloading the `benchmark-all` resource forces all Go services to be benchmarked again.
- Go services following the typical pattern *do not require a Dockerfile or a Kubernetes manifest*. Those are generated automatically at runtime and injected into Tilt.

Notable screw-ups:

- The `frontend` service stopped working and I couldn't find what's wrong. Hälp?

### CODE REVIEW

File: fast/1.0-basic/Tiltfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+k8s_yaml([
+    'glitch/k8s.yaml',
+    'storage/k8s.yaml',
+    'muxer/k8s.yaml',
+])
+
+k8s_resource("muxer", port_forwards="8080")
+
+local_resource('update', "sed -re 's/time.Unix\(0, ([0-9]*)\)/time.Unix\(0, '\"$(($(date +%s%N)))\"'\)/' -i glitch/main.go")
```
Comment (jazzdan) [inline fast/1.0-basic/Tiltfile:13]: We repeat this a lot, can we put this in a shared Tiltfile that is include dfrom all of the other ones?

File: fast/1.0-basic/glitch/main.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	"github.com/pkg/errors"
+	"github.com/sug0/go-glitch"
+	"github.com/windmilleng/enhance/render/api"
+
+	_ "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// syntax described here: https://github.com/sug0/go-glitch/blob/master/res/cool.txt
+// various equations here: https://github.com/sug0/go-glitch/blob/master/res/cool.txt
+var glitchExprStr = "b ^ (r | (s : (x # B)))"
```
Comment (jazzdan) [inline fast/1.0-basic/glitch/main.go:23]: This could be a const I think, right?

Comment (ellenkorbes) [reply fast/1.0-basic/glitch/main.go]: Good catch!

**Comment by ellenkorbes** [comment]
@jazzdan 

> Do we want to be comitting all of these cache files?
By cache files do you mean the intermediate steps or is there some cache I'm missing somewhere?

> What's hot sauce?
That's hot/live reloading the source code into a running container.

> is it worth it to show all these intermediate optimization steps separately in what is going to be a sample repo? It it unnecessary complexity?
This repo is not only home to the sample project, but also the repo showing the stages presented in the talk. That's why all the steps are there. I separated them into different folders at the root to try and avoid that confusion. Will add a README as well.

You could argue these should be in different repos, and I agree, but while this project is still changing—new frontend coming soon—I'd rather keep it all together.

Does that make sense?

File: fast/1.0-basic/Tiltfile (hunk 1 above)
Comment (ellenkorbes) [reply fast/1.0-basic/Tiltfile:13]: The different versions in the `fast/` directory are snapshots at different points in time. At any given moment, that's the only instance of it, so I'd feel weird abstracting something away that for the audience effectively only appears once.

**Comment by jazzdan** [comment]
@ellenkorbes as for cache I'm referring to files like this https://github.com/windmilleng/enhance/pull/22/files#diff-efdfc5d142698393742641d02a182ce8

and yeah putting it all in one place sounds good to me! Just wanted to make sure we'd thought it hrough and it sounds like we did :)

**Comment by ellenkorbes** [comment]
That's a good question. I think for didactic purposes it should stay.

Merging for now.

**Pull request merged**